app/main.py

```diff
@@ -1,2 +1,108 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._length: int = 0
+        self._hash_table: list = [None] * 8
+        self._hash_size: int = 8
+        self._resize_factor: float | int = 2 / 3
+        self._resize_multiplier: int = 2
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self._hash_size_check_increase()
+        key_hash_position = self._find_key_position(key)
+
+        while self._hash_table[key_hash_position]:
+            if self._hash_table[key_hash_position][0] == key:
+                self._hash_table[key_hash_position][1] = value
+                return print(f"value with key '{key}' were rewritten")
+            key_hash_position = (key_hash_position + 1) % self._hash_size
+
+        self._hash_table[key_hash_position] = [key, value]
+        self._length += 1
+        print(f"'key:value = {key}:{value}' "
+              f"is set to {key_hash_position} index in the Dictionary")
+
+    def __getitem__(self, item: Hashable) -> Any | None:
+        key_hash_position = self._find_key_position(item)
+
+        # find item by key_hash_position and if it has the right key to it
+        while self._hash_table[key_hash_position]:
+            if self._hash_table[key_hash_position][0] == item:
+                return self._hash_table[key_hash_position][1]
+            key_hash_position = (key_hash_position + 1) % self._hash_size
+
+        raise KeyError("Key is not found in the Dictionary")
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __repr__(self) -> str:
+        result = [f"{item[0]}:{item[1]}"
+                  for item in self._hash_table
+                  if item is not None]
+        return "{" + ", \n".join(result) + "}"
+
+    def __delitem__(self, key: Hashable) -> None:
+        key_hash_position = self._find_key_position(key)
+
+        while self._hash_table[key_hash_position]:
+            if self._hash_table[key_hash_position][0] == key:
+                self._hash_table[key_hash_position] = None
+                print(f"removed: {key} from position {key_hash_position}")
+                self._length -= 1
+                return
+            key_hash_position = (key_hash_position + 1) % self._hash_size
+
+        raise KeyError("Key is not found in the Dictionary")
+
+    def pop(self, key: Hashable) -> Any:
+        result = self.__getitem__(key)
```

Comment:
use syntactic sugar here since you have these methods implemented:
e.g. del self[key]